### PR TITLE
Create Blog “open-sourcing-workshops-on-demand-part4-managing-the-backend”

### PR DESCRIPTION
**Pull request by pattdepanick**
Automatically generated by Netlify CMS

### CODE REVIEW

**Review by choukden — CHANGES_REQUESTED** [review]
Hi Fredo,

Thank you very much for your new contribution.

Suggestions with changes discussed together to add tags, change the heading level from level 4 to level 3 and from level 5 to level 4, and finally try to manage the bullet item lists and sub-bullet lists towards the end of the blog post. And finally change the date of the post to today.

I will ask Dale to go through the editorial review.

THANKS AGAIN!!!

Denis

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
+---
+title: "Open Sourcing Workshops-on-Demand part 4: Manage the Backend"
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:2]: If you can, please modify the capitalization of the title to:
Open sourcing Workshops-on-Demand Part 4: Manage the backend

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+date: 2023-07-18T09:28:35.992Z
+author: Frederic Passeron
+authorimage: /img/frederic-passeron-hpedev-192.jpg
+disable: false
+tags:
+  - " opensource"
+  - workshops-on-demand
+---
+I﻿n previous articles of this series dedicated to the [open sourcing of our Workshops-on-Demand project](https://developer.hpe.com/blog/willing-to-build-up-your-own-workshops-on-demand-infrastructure/), I covered the reasons why we open sourced  the project, how we did it. I also explained in details how you could install your own Workshops-on-Demand backend server. I also took the time to detail the automation that was hosted on this backend server. Today, I plan to describe to you the management of this backend server. What is often refered to as day2 operations.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:11]: change "why we open sourced the project, how we did it." to:
"why we open sourced the project and how we did it."

This is interesting, Fred... right here, I see "I also explained in details how you could..." and I was going to recommend modifying that somewhat, but in the preview, it says "I also took the time to detail the..."

I'm not sure which is the most current. My preference is the "I also took the time to detail the automation..."

The last sentence should be modified to read:
"This is what is often referred to as Day 2 operations." (please note the spelling of "referred")

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+disable: false
+tags:
+  - " opensource"
+  - workshops-on-demand
+---
+I﻿n previous articles of this series dedicated to the [open sourcing of our Workshops-on-Demand project](https://developer.hpe.com/blog/willing-to-build-up-your-own-workshops-on-demand-infrastructure/), I covered the reasons why we open sourced  the project, how we did it. I also explained in details how you could install your own Workshops-on-Demand backend server. I also took the time to detail the automation that was hosted on this backend server. Today, I plan to describe to you the management of this backend server. What is often refered to as day2 operations. 
+
+O﻿nce up and running, the backend server main purpose is to deliver workshops-on-Demand. But to do so, it may require updates, upgrades, new kernels for the Jupyterhub server. If new workshops are created, this means new jinja templates for related workshops' scripts (`create<WKSHP>.sh`, `cleanup<WKSHP>.sh`, `reset<WKSHP>.sh` among others). This also means new variables files. And obviously, these templates and variables will need to be taken into account by scripts and notebooks. Some tasks handle all of this. And we will see now how.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:13]: Modify first sentence to read:
Once up and running, the main purpose of the backend server is to deliver the Workshops-on-Demand.

In the second sentence, modify the second part to read:
"...updates, upgrades, and/or new kernels for the JupyterHub server."

Third sentence should read:
"If new workshops are created, this means you'll need new Jinja templates for the related workshops' scripts (i.e. 'create<WKSHP>.sh', 'cleaup<WKSHP>.sh', 'reset<WKSHP>.sh', among others)." (note the comma prior to "among others")

Should it be "new variables files" or "new variable files"? (my guess would be the latter)

Please change "And we will see now how." to:
"And that's what I'll show you now." (remember... avoid the Royal We... you and I may be aristocracy, but it's not nice to flaunt it in the reader's face, lol)

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+#### B﻿ackend server management:
+
+If you take a look at the file structure of the `wod-backend` directory, you will discover that we did our best to sort things properly depending on their relationship to system  or workshops.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:17]: To avoid the Royal We here (though I know you're referring to an actual "we" here), try this:

"... you will discover that the team did its best to sort things properly..."

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+##### C﻿ontent of the backend server:
+
+S﻿imple tree view of the wod-backend directory:
+
+![](/img/wod-blogserie2-tree1.png "Tree view of wod-backend directory")
+
+T﻿he `ansible` folder contains all the necessary playbooks and variables files to support the main functions of the backend server. It provides playbooks for a minimal installation of the server or appliances, setup of the servers (backend, frontend, api-db),of appliances or workshops as well as maintenance tasks.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:25]: hmmm... maybe you did mean "variables". Or, would "variable files" work here?

Check the comma right after the api-db). I think you need a period after it (just prior to "of appliances")

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+
+A﻿t the root of this directory can be found:
+
+`C﻿heck*.yml playbooks`: These playbooks are used to perform checks on the different systems. These checks ensure that this a compliant WoD system by checking Firewall rules and many other things. We will see this a bit later in more details.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:29]: "Firewall" should be "firewall"
As opposed to saying "We will see this..." say "You will see this..."

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+
+`C﻿opy_folder.yml`: This is historically one of very first playbook we used and therefore a very important one. It performs the necessary actions to deploy, personnalize (by substituting ansible variables) the selected notebook to the appropriate student home folder.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:31]: Modify this section "This is historically..." to:
"Historically, this is one of the very first playbooks the team used, and therefore, it is very important to me."

Change the second sentence to remove the comma and use the word "and" instead so it reads:
"It performs the necessary actions to deploy and personnalize (by substituting Ansible variables) the selected notebook to the appropriate student home folder." (note capitalization of "Ansible")

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+
+`c﻿ompile_scripts.yml`: Should you need to hide from the student a simple api call that is made on some private endpoint with non shareable data (credentials for instance), this playbook will make sure to compile it and create a executable file allowing it to happen.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:33]: place a hyphen between "non" and "shareable" so it reads: "non-sharable data"

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+
+`C﻿heck*.yml playbooks`: These playbooks are used to perform checks on the different systems. These checks ensure that this a compliant WoD system by checking Firewall rules and many other things. We will see this a bit later in more details.
+
+`C﻿opy_folder.yml`: This is historically one of very first playbook we used and therefore a very important one. It performs the necessary actions to deploy, personnalize (by substituting ansible variables) the selected notebook to the appropriate student home folder.
+
+`c﻿ompile_scripts.yml`: Should you need to hide from the student a simple api call that is made on some private endpoint with non shareable data (credentials for instance), this playbook will make sure to compile it and create a executable file allowing it to happen. 
+
+`d﻿istrib.yml`: This playbook retrieves the distribution name and version from the machine it is run on.
+
+`install_*.yml`: These playbooks take care of installing the necessary packages needed by the defined type (frontend, backend, api-db, base-system or even appliance)
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:37]: Add a period to the end here.

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+
+`setup_*.ym`: There are several types of setup playbooks in this directory. 
+
+* `setup_WKSHP-*.yml`: These playbooks are responsible for preparing a base appliance for a given workshop by adding and configuring the necessary packages or services related to the workshop.
+* `s﻿etup_appliance.yml`: this playbook is used to perform the base setup for a JupyterHub environment server or appliance. It includes setup_base_appliance.yml playbook.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:42]: Capitalize "This" for "This playbook is used..."

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+* `s﻿etup_base_appliance`: takes care of setting the minimal requierements for an appliance. it includes therefore `install_base_system.yml` playbook. On top of it, it creates and configure the necessary users.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:43]: Just prior to "takes care", add "This" so it reads "This takes care of...

In the second sentence, capitalize "it" for "It includes"  Take out the word "therefore"

In the last sentence, change "configure" to "configures" (subject/verb tense agreement)

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+* `s﻿etup_docker_based_appliance.yml`: Quite self explanatory ? it performs setup tasks to enable docker on a given appliance.
+
+It also hosts the `inventory` file describing the role of jupyterhub servers. Place your jupyterhub machine (FQDN) in a group used as PBKDIR namerole
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:46]: Change "jupyterhub" to "JupyterHub" in both instances. 
Place a period at the end of the last sentence.

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 13]
```diff
@@ -0,0 +1,242 @@
...
+
+```shellsession
+#
+# Place to your jupyterhub machine (FQDN) in a group used as PBKDIR name
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:50]: Same here... JupyterHub and a period.

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 14]
```diff
@@ -0,0 +1,242 @@
...
+#
+[production]
+127.0.0.1  ansible_connection=localhost
+```
+
+T﻿he `conf` folder hosts configuration files in a jinja format. Once expanded, the resulting files will be used by relevant workshops. I will explain in a future article all the steps and requirements to create a workshop.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:56]: pretty sure Jinja should have an initial capital letter, as it's the proper name of the thing.

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 15]
```diff
@@ -0,0 +1,242 @@
...
+
+A﻿s part of the refacturing work to open source the project, we reaaranged the different scripts locations. We have created an install folder to handle the different installation scripts either from a Jupyterhub 's perpective or from an appliance 's standpoint too.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:58]: Do you mean "refactoring"? I'm not familiar with "refacturing"
Place an apostrophe after "scripts" since it's possessive. "...different scripts' locations."
JupyterHub has a capital H in it. It also looks like you have an extra space between the "b" and the "s". Similarly for the word "appliance".
Place a comma just prior to "too."

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 16]
```diff
@@ -0,0 +1,242 @@
...
+
+We separated the workshops related scripts from the system ones. When one creates a workshop, one needs to provide a series of notebooks and in some cases some scripts to manage the creation, setup of a related appliance along with additional scripts to manage its lifecycle in the overall workshops-on-Demand architecture (Create, Cleanup, Reset scripts at deployment or Cleanup times). These scripts need to be located in the `scripts` folder. On the other hand, the system scripts are located in the `sys` folder.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:60]: "workshops related" should be "workshops' related" as it is also posessive.
Change "...to manage the creation, set up of a..." to: "...to manage the creation and set up of a..."

"workshops-on-Demand" should be "Workshops-on-Demand"

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 17]
```diff
@@ -0,0 +1,242 @@
...
+
+![](/img/tree-wkshop2.png "Tree view of the sys directory")
+
+T﻿his directory hosts important configuration files for both the system and Jupyterhub. You can see for instance `fail2ban` configuration files. Some jinja templates are present here too. These templates will be expanded trough the `deliver` mechanism allowing the creation of files customized with ansible variables. All the wod related tasks are prefixed with wod for better understanding and ease of use.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:64]: JupyterHub (capital H)
Jinja
Place a comma prior to "too" in the third sentence.
The word "through" is misspelled in the 4th sentence. Capitalize "Ansible"

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 18]
```diff
@@ -0,0 +1,242 @@
...
...me Jupyterhub kernel needs like `wod-build-evcxr.sh.j2` that aims at creating a script allowing the rust kernel installation. Some other templates are related to the system and Jupyterhub. `wod-kill-processes.pl.j2` has been created after discovering the harsh reality of online mining...In a ideal world, I would not have to explain further as the script would not be needed. Unfortunately, this is not the case, when one offers access to some hardware freely online, he can expect sooner or later to  see his original and pure idea to be highjacked...Let's say that you want to provide some AI/ML 101 type of workshops, you may consider providing servers with some GPUs. Any twisted minded cryptominer discovering your ressources will definitely think he hits the jackpot! This little anecdot actually happened to us and not only on GPU based servers, some regular servers got hit as well. We found out that performance on some servers became very poor and when looking into it, we found some scripts that were not supposed to run there...As a result, we implemented monitors to check load on our server and make sure that to  kill any suspicious process before kicking out the misbehaving student.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:66]: What do you mean by "They"? The Jina templates? The wod-related tasks? It's best to spell it out at the beginning of a paragraph so the reader is clear on what the subject is here.

JupyterHub kernel.

Replace the elipsis (three dots) with a period.

Modify the section that reads "Unfortunately..." to:
"Unfortunately, this is not the case. When one offers access to hardware freely online, sooner or later, he can expect to see his original idea to be hyjacked.  

This is a very long paragraph and it would be easier to read if it were broken out into two paragraphs, but I'm not quite sure where the best break is. I'll leave you to decide that.

Don't use the elipsis. You have it at least 3 times in this paragraph. Just go with simple sentences, ending them each with a period.

It might be good to start the second paragraph right after "highjacked". Then, start the next line with a modified version of what you have:
"Let's say that you want to provide some AI/ML 101-type of workshops. As part of this, you may consider providing servers with GPUs. Any twisted-minded cryptominer discovering your resources will definitely think that he's hit the jackpot!"  (note that "resources" had a typo)

The last sentence should be modified to read:
"As a result, we implemented monitors to check the load on our server and made sure we killed any suspicious processes before kicking out the misbehaving student."  [I love this, by the way... great story!]

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 19]
```diff
@@ -0,0 +1,242 @@
...
+
+`wod-test-action.sh.j2` is another interesting template that will create a script that we use for testing. This script mimics the procmail API and actually enables you to test the complete lifecycle of a workshop from deployment to cleanup or reset.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:68]: Instead of "that we use for testing" maybe try "that can be used for testing."

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 20]
```diff
@@ -0,0 +1,242 @@
...
+
+```shellsession
+wodadmin@server:/usr/local/bin$ ./wod-test-action.sh
+Syntax: wod-test-action.sh <CREATE|CLEANUP|RESET|PURGE|PDF|WORD> WKSHOP [MIN[,MAX]
+ACTION is mandatory
+wodadmin@server:/usr/local/bin$
+```
+
+I﻿t requires the verb, the workshop's name and the student id. Using the script, one does not need to provide participant id.  The script is run locally on the Jupyterhub server.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:77]: JupyterHub

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 21]
```diff
@@ -0,0 +1,242 @@
...
+Sending a mail to CREATE student 121 for workshop WKSHP-API101
+220 server.xyz.com ESMTP Postfix (Ubuntu)
+250 2.1.0 Ok
+250 2.1.5 Ok
+354 End data with <CR><LF>.<CR><LF>
+250 2.0.0 Ok: queued as 9749E15403AB
+221 2.0.0 Bye
+```
+
+I﻿n order to retrieve the result of the script, you simply need to run a `tail` command
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:96]: place a period at the end of this sentence

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 22]
```diff
@@ -0,0 +1,242 @@
...
+ Subject: CREATE 121 0
+  Folder: /home/wodadmin/wod-backend/scripts/procmail-action.sh CREATE       11
++ source /home/wodadmin/wod-backend/scripts/wod.sh
+....
++ echo 'end of procmail-action for student 121 (passwd werty123) with workshop WKSHP-API101 with action CREATE at Fri Mar  3 09:11:39 UTC 2023'
+```
+
+T﻿he very last line of the trace will provide you with the credentials necessary to test your workshop. 
+
+T﻿here are two types of activities that can occur on the backend server: punctual or regular. The punctual activity is one that is performed once every now and then. The regular one is usually set up on the backend server as a cron job. Sometimes however, one of these cron tasks can be forced manually if necessary. One the main scheduled task is the `deliver` task. I will explain it later on in this chapter. I will start now by explaining an important possible punctual task, the update of the backend server.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:112]: I don't understand what you mean by "One the main scheduled task is the 'deliver' task." I think the word "One" may be incorrect or something is missing?

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 23]
```diff
@@ -0,0 +1,242 @@
...
+
+#### U﻿pdate of the backend server:
+
+T﻿he backend server hosts all the necessary content for delivering workshops: it implies notebooks and scripts and playbooks to deploy and personalize them. It also hosts some services that are needed by the overall architecture solution (Jupyterhub, Procmail, Fail2ban among others).
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:116]: "...workshops: it implies"  should this actually be "it supplies"?
JupyterHub

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 24]
```diff
@@ -0,0 +1,242 @@
...
+
+S﻿ervices are installed once and for all at the installation time. These services may evolve over time. One may need to update the jupyterhub application to fix a bug or get new features. In the same fashion, you may consider bumping from one python version to a new major one. If you are willing to update these services or add new ones, you will need to update the relevant installation playbooks in `wod-backend/ansible` directory.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:118]: JupyterHub application
Python has a capital "P"

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 25]
```diff
@@ -0,0 +1,242 @@
...
+
+h﻿ere is a small extract of the `install_backend.yml` playbook: Full version [here](https://github.com/Workshops-on-Demand/wod-backend/blob/main/ansible/install_backend.yml)
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:120]: Start the sentence with a capital letter: "Here is a small..."

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 26]
```diff
@@ -0,0 +1,242 @@
...
+      apt_repository:
+        repo: deb [arch=amd64] https://repo.anaconda.com/pkgs/misc/debrepo/conda stable main
+        state: present
+      when:
+       - ansible_distribution == "Ubuntu"
+```
+
+P﻿ossible Use Cases:
+
+* U﻿pgrade to a newer version of Jupyterhub
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:170]: JupyterHub in this line and the next

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 27]
```diff
@@ -0,0 +1,242 @@
...
+For e.g:
+
+* Kubectl client  
+* T﻿erraform client
+* P﻿owerShell module  
+* P﻿ython Library
+
+You will start by moving to your public backend forked repository and apply the necessary changes before committing and push locally. 
+
+Then you will perform a merge request with the main repository. We plan to integrate here  in a proper CICD (continuous integration continous development) pipeline to allow a vagrant based test deployment. Whenever someone performs a merge request on the main repo, the test deployment task kicks in and deploy a virtual backend server on which the new version of the installation process is automatically tested. When successful, the merge request is accepted. Once merged, you will need to move to your backend server and perform git remote update and git rebase on the wod-backend directory. Once done, you will then be able to perform the installation process.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:185]: In the middle of the 3rd sentence, it should read "the test deployment task kicks in and deploys..." (s at the end of deploy)

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 28]
```diff
@@ -0,0 +1,242 @@
...
+* Install the pkg update script
+* Setup `crontab` for daily pkg security update
+* Deliver create/reset/setup scripts as ansible template for variable expansion
+* Install utility scripts
+* Deliver the system scripts (`cleanup-processes.sh.j2`)
+* Installation of the cleanup-processes script
+* Setup weekly cleanup processes task
+* Enable WoD service
+* Test private tasks YAML file
+* Call private tasks if available. We perform private part before users management to allow interruption of the deliver script during normal operations - waiting till end of users management can take hours for 2000 users. Potential impact: private scripts are run before users creation, so may miss some part of setup.
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:227]: Do you mean "We perform **the** private part..."?

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 29]
```diff
@@ -0,0 +1,242 @@
...
+* U﻿ser Management:
+
+  * Remove existing jupyterhub users
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:230]: JupyterHub in lines 230, 234, 235, and 236

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 30]
```diff
@@ -0,0 +1,242 @@
...
+  * Remove Linux users and their home directory
+  * Ensure dedicated students groups exist
+  * Ensure Linux students users exists with their home directory
+  * Ensure jupyterhub students users exist
+  * Setup ACL for students with jupyterhub account
+  * Setup default ACL for students with jupyterhub account
+
+A﻿ similar set of scripts exist for the different parts of the solution ([check_api-db.yml](https://github.com/Workshops-on-Demand/wod-backend/blob/main/ansible/check_api-db.yml) for api-db server, [check_frontend.yml ](https://github.com/Workshops-on-Demand/wod-backend/blob/main/ansible/check_frontend.yml)for frontend server for instance).
+
+You should have now a better understanding of the maintenance tasks associated to the backend server. Similar actions are available for the other components of the project. Checking tasks have been created for the frontend and api-db server. Having nowostly covered all the subjects related to the backend server from an infrastructure standpoint, it is high time to discuss the content part. In my next blog, I plan to describe the workshop creation process.  Time to understand how to build up some content for the jupyterhub server!
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:240]: Swap placement of "have" and "now" so it reads "You should now have a better..."

In the 4th sentence, there's a weird word "nowostly". I think you meant "now mostly"

JupyterHub server!

File: content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md [hunk 31]
```diff
@@ -0,0 +1,242 @@
...
+
+If we can be of any help in clarifying any of this, please reach out to us on [Slack](https://slack.hpedev.io/). Please be sure to drop back at [HPE DEV](https://developer.hpe.com/blog) for a follow up on this. Check out also the Hack Shack for new [workshops](https://developer.hpe.com/hackshack/workshops)! Willing to collaborate with us? Contact us and let's build together some more workshops! Stay tuned!
```
Comment (drensing) [inline content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md:242]: I suggest changing "...to drop back at..." to "... to check back at..."
Then, in the next sentence, I'd reword it to "Also, don't forget to check out the Hack Shack for new..."
Second to last sentence, modify to "Contact us so we can work together to build more workshops!"

**Review by drensing — COMMENTED** [review]
Fred, this was lovely! You seemed to have taken more time creating this one and were more careful on the language you used. 

Nice job! I do, of course, have edits, but not nearly as many as usual.

All the best,
Dale
